v-next/hardhat/src/internal/builtin-plugins/solidity-test/reporter.ts

```diff
@@ -117,27 +122,60 @@ export async function* testReporter(
     for (const [index, failure] of failures.entries()) {
       yield `\n${chalk.bold(chalk.red(`Failure (${index + 1})`))}: ${failure.name}\n`;
 
-      if (failure.decodedLogs !== undefined && failure.decodedLogs.length > 0) {
-        yield `Decoded Logs${chalk.grey(`: ${failure.decodedLogs.join("\n")}\n`)}`;
+      const stackTrace = failure.stackTrace();
+      if (
+        stackTrace !== undefined &&
+        stackTrace !== null &&
+        stackTrace.length > 0
+      ) {
+        const stackTraceMessage = getMessageFromLastStackTraceEntry(
+          stackTrace[stackTrace.length - 1],
+        );
+
+        const stackTraceStack: string[] = [];
+        for (const entry of stackTrace.reverse()) {
+          const callsite = encodeStackTraceEntry(entry);
+          if (callsite !== undefined) {
+            stackTraceStack.push(`  at ${callsite.toString()}`);
+          }
+        }
+
+        if (stackTraceMessage !== undefined || stackTraceStack.length > 0) {
+          yield `${stackTraceMessage ?? "Stack trace:"}\n${chalk.grey(stackTraceStack.join("\n"))}\n`;
+        }
+      }
+
+      if (
+        failure.decodedLogs !== undefined &&
+        failure.decodedLogs !== null &&
+        failure.decodedLogs.length > 0
+      ) {
```

Comment:
I added extra `!== null` checks to all the properties of `TestResult` we inspect because I noticed they sometimes reach us as `null`s even though that's not reflected in their types (I noticed that for `failure.reason` but decided to apply the checks to everything just in case).

Reply:
cc @agostbiro

Reply:
I can repro it in EDR integration tests, but not in a standalone NAPI-rs project. I created an [issue](https://github.com/NomicFoundation/edr/issues/796) and set it aside for now until after the alpha release.